diff --git a/imconfig.h b/imconfig.h
index 6e4c743..3403873 100644
--- a/imconfig.h
+++ b/imconfig.h
@@ -13,7 +13,15 @@
 //-----------------------------------------------------------------------------
 
 #pragma once
-
+#define ImDrawIdx unsigned int
+#define IMGUI_OVERRIDE_DRAWVERT_STRUCT_LAYOUT struct ImDrawVert\
+{\
+    ImU32   textureID;\
+    ImVec2  pos;\
+    ImU32   col;\
+    ImVec2  uv;\
+    ImVec2  __padding__;\
+};
 //---- Define assertion handler. Defaults to calling assert().
 // If your macro uses multiple statements, make sure is enclosed in a 'do { .. } while (0)' block so it can be used as a single statement.
 //#define IM_ASSERT(_EXPR)  MyAssert(_EXPR)
