diff --git a/imconfig.h b/imconfig.h
index 6e4c743..05d9184 100644
--- a/imconfig.h
+++ b/imconfig.h
@@ -13,6 +13,16 @@
 //-----------------------------------------------------------------------------

 #pragma once
+#define IMGUI_DISABLE_DEMO_WINDOWS
+#define ImDrawIdx unsigned int
+#define IMGUI_OVERRIDE_DRAWVERT_STRUCT_LAYOUT struct ImDrawVert\
+{\
+    ImU32   textureID;\
+    ImVec2  pos;\
+    ImU32   col;\
+    ImVec2  uv;\
+    ImVec2  __padding__;\
+};

 //---- Define assertion handler. Defaults to calling assert().
 // If your macro uses multiple statements, make sure is enclosed in a 'do { .. } while (0)' block so it can be used as a single statement.
