diff --git a/src/win32_window.c b/src/win32_window.c
index e6a9496..be4f36d 100644
--- a/src/win32_window.c
+++ b/src/win32_window.c
@@ -63,6 +63,9 @@ static DWORD getWindowStyle(const _GLFWwindow* window)

 // Returns the extended window style for the specified window
 //
+#ifndef WS_EX_NOREDIRECTIONBITMAP
+ #define WS_EX_NOREDIRECTIONBITMAP 0x00200000L
+#endif
 static DWORD getWindowExStyle(const _GLFWwindow* window)
 {
     DWORD style = WS_EX_APPWINDOW;
@@ -70,6 +73,9 @@ static DWORD getWindowExStyle(const _GLFWwindow* window)
     if (window->monitor || window->floating)
         style |= WS_EX_TOPMOST;

+    if (getenv("RAWRBOX_DCOMP_TRANSPARENT"))
+        style |= WS_EX_NOREDIRECTIONBITMAP;
+
     return style;
 }