diff --git a/include/ozz/animation/runtime/animation.h b/include/ozz/animation/runtime/animation.h
index 27f979d..300db33 100644
--- a/include/ozz/animation/runtime/animation.h
+++ b/include/ozz/animation/runtime/animation.h
@@ -51,6 +51,13 @@ struct Float3Key;
 struct QuaternionKey;
 }  // namespace internal
 
+
+enum AnimationType : uint32_t {
+  UNKNOWN = 0,
+  VERTEX = 1,
+  SKELETON = 2
+};
+
 // Defines a runtime skeletal animation clip.
 // The runtime animation data structure stores animation keyframes, for all the
 // joints of a skeleton. This structure is usually filled by the
@@ -63,6 +70,8 @@ struct QuaternionKey;
 // time, then by track number.
 class OZZ_ANIMATION_DLL Animation {
  public:
+  AnimationType type = AnimationType::UNKNOWN;
+
   // Builds a default animation.
   Animation();
 
diff --git a/include/ozz/animation/runtime/skeleton.h b/include/ozz/animation/runtime/skeleton.h
index 4fdc22a..dde2c2d 100644
--- a/include/ozz/animation/runtime/skeleton.h
+++ b/include/ozz/animation/runtime/skeleton.h
@@ -33,6 +33,9 @@
 #include "ozz/base/platform.h"
 #include "ozz/base/span.h"
 
+#include "vector"
+#include "array"
+
 namespace ozz {
 namespace io {
 class IArchive;
@@ -79,6 +82,8 @@ class OZZ_ANIMATION_DLL Skeleton {
     kNoParent = -1,
   };
 
+  std::vector<std::array<float, 16>> inverseBindMatrices;
+
   // Builds a default skeleton.
   Skeleton();
 
