diff --git a/CMakeLists.txt b/CMakeLists.txt
index 68b5343..a3e2aa0 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -210,9 +210,9 @@ else()
     #     -DCURL_ZLIB=OFF
     # * from CMake script:
     #     SET(CURL_ZLIB OFF CACHE STRING "" FORCE)
-    if (CURL_ZLIB OR CURL_ZLIB STREQUAL AUTO OR NOT DEFINED CACHE{CURL_ZLIB})
-        include(cmake/zlib_external.cmake)
-    endif()
+    #if (CURL_ZLIB OR CURL_ZLIB STREQUAL AUTO OR NOT DEFINED CACHE{CURL_ZLIB})
+    #    include(cmake/zlib_external.cmake)
+    #endif()
 
     if (CPR_ENABLE_CURL_HTTP_ONLY)
         # We only need HTTP (and HTTPS) support:
