diff --git a/CMakeLists.txt b/CMakeLists.txt
index 75b160c..339c803 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -482,7 +482,7 @@ IF(ASSIMP_HUNTER_ENABLED)
 ELSE()
   # If the zlib is already found outside, add an export in case assimpTargets can't find it.
   IF( ZLIB_FOUND )
-    INSTALL( TARGETS zlib zlibstatic
+    INSTALL( TARGETS zlib
         EXPORT "${TARGETS_EXPORT_NAME}")
   ENDIF()
 
diff --git a/include/assimp/ZipArchiveIOSystem.h b/include/assimp/ZipArchiveIOSystem.h
index 9f3a478..0589e44 100644
--- a/include/assimp/ZipArchiveIOSystem.h
+++ b/include/assimp/ZipArchiveIOSystem.h
@@ -55,6 +55,7 @@ OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
 #include <assimp/IOStream.hpp>
 #include <assimp/IOSystem.hpp>
+#include <zlib.h>
 
 namespace Assimp {
 
